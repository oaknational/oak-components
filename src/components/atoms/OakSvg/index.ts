export { OakSvg } from "./OakSvg";
export type { OakSvgProps, OakSvgNames } from "./OakSvg";
