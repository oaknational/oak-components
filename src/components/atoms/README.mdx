# Atoms

## Rules for components in this folder

- Can import other atoms
- Imports from molecules and organisms are forbidden!
- Generic
- Unstyled
- Avoid embedded logic and state
- No sub-components
  (eg. Box)
