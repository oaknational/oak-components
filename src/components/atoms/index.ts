export * from "./OakBox";
export * from "./OakFlex";
export * from "./OakSpan";
export * from "./OakForm";
export * from "./OakImage";
export * from "./OakTypography";
export * from "./OakHeading";
export * from "./OakP";
export * from "./OakUL";
export * from "./OakOL";
export * from "./OakLI";
export * from "./OakLabel";
export * from "./OakIcon";
export * from "./OakGrid";
export * from "./OakGridArea";
export * from "./OakThemeProvider";
export * from "./OakMaxWidth";
export * from "./OakCloudinaryImage";
export * from "./OakKbd";
export * from "./OakGlobalStyle";
export * from "./OakAnchorTarget";
export * from "./OakFieldSet";
