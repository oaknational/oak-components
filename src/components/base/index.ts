export * from "./OakBox";
export * from "./OakFlex";
export * from "./OakBaseCard";
export * from "./OakSpan";
export * from "./OakForm";
export * from "./OakImage";
export * from "./OakTypography";
export * from "./OakHeading";
export * from "./OakP";
export * from "./OakUL";
export * from "./OakOL";
export * from "./OakLI";
export * from "./OakLabel";
