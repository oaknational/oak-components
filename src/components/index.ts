export * from "./base";
// export * from "./ui"; // uncomment this line once there are components in this folder
// export * from "./integrated"; // uncomment this line once there are components in this folder
