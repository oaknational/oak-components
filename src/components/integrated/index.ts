export * from "./OakQuizCheckBox";
export * from "./OakQuizTextInput";
export * from "./OakQuizRadioButton";
export * from "./OakQuizHint";
export * from "./OakQuizBottomNav";
export * from "./OakLessonTopNav";
export * from "./OakQuizFeedback";
export * from "./OakQuizCounter";
