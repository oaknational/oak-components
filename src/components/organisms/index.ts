export * from "./pupil";
export * from "./teacher";
export * from "./OakSubjectIcon";
export * from "./OakPrimaryNavItem";
export * from "./OakPrimaryNav";
export * from "./OakCookieBanner";
export * from "./OakCookieSettingsModal";
export * from "./OakCookieConsentProvider";
export * from "./OakCookieConsent";
