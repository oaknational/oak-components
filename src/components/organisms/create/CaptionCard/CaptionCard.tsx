import React from "react";
import styled, { css } from "styled-components";

import { getTimeText } from "@/components/organisms/create/utils";
import { OakBox, OakFlex, OakIcon, OakSpan } from "@/components/atoms";
import { InternalCheckBoxLabelHoverDecor } from "@/components/atoms/InternalCheckBoxLabel";
import { parseColor } from "@/styles/helpers/parseColor";
import { parseDropShadow } from "@/styles/helpers/parseDropShadow";
import { InternalButton } from "@/components/atoms/InternalButton";
import { OakCheckBox } from "@/components/molecules";
import { InternalLink } from "@/components/molecules/InternalLink";

// Converted to styled-component so it can be used in '&:checked:not(:disabled) + ${StyledOakIcon}' to change svg color.
const StyledOakIcon = styled(OakIcon)``;

interface StyledFlexBoxWrapperProps {
  $minHeight: string;
  $position: string;
  $borderRadius: string;
  $pa: string;
  $gap: string;
  $flexDirection: string;
  $width: string;
  $highlighted?: boolean;
  $disabled?: boolean;
}

const StyledFlexBox = styled(OakFlex)<StyledFlexBoxWrapperProps>`
  &:has(input:not(:disabled)) {
    cursor: default;
  }

  &:has(input:disabled) {
    pointer-events: none;
    cursor: none;
  }

  &:hover:has(input:not(:disabled)) ${InternalCheckBoxLabelHoverDecor} {
    text-decoration: underline;
  }

  &:focus-within {
    box-shadow: ${parseDropShadow("drop-shadow-centered-lemon")},
      ${parseDropShadow("drop-shadow-centered-grey")};
  }

  ${(props) =>
    props.$disabled &&
    css`
      cursor: not-allowed;
      background-color: ${parseColor("bg-neutral-stronger")};
    `}

  ${(props) =>
    !props.$disabled &&
    props.$highlighted &&
    css`
      background-color: ${parseColor("bg-decorative3-very-subdued")};
      border-color: ${parseColor("bg-decorative3-main")};
      &:hover {
        background-color: ${parseColor("bg-decorative3-subdued")};
      }
    `}
  ${(props) =>
    !props.$disabled &&
    !props.$highlighted &&
    css`
      &:hover {
        background-color: ${parseColor("bg-decorative3-subdued")};
      }
    `}
`;

export interface CaptionCardProps {
  captionId: string;
  videoTitle: string;
  lessonUid: string;
  videoType: "lesson";
  lastUpdated: string;
  lastEdited?: string;
  checked: boolean;
  highlighted?: boolean;
  disabled?: boolean;
  onCheckChanged?: (event: React.ChangeEvent<HTMLInputElement>) => void;
  onLessonUidClick?: () => void;
  onEditClick?: () => void;
  "data-testid"?: string;
}

/**
 * A caption card with links to the associated lesson and rev edit page.
 *
 * ## Events
 * The following callbacks are available for tracking checkbox events:
 *
 * ### onCheckChanged
 *  onCheckChanged?: (event: React.ChangeEvent<HTMLInputElement>) => void;
 *
 */
export const CaptionCard = (props: CaptionCardProps) => {
  const {
    checked,
    onCheckChanged,
    onLessonUidClick,
    onEditClick,
    captionId,
    videoTitle,
    lessonUid,
    videoType,
    lastUpdated,
    lastEdited,
    highlighted = false,
    disabled = false,
    "data-testid": dataTestId = "caption-card",
  } = props;

  return (
    <StyledFlexBox
      data-testid={dataTestId}
      $minHeight={"all-spacing-8"}
      $position={"relative"}
      $borderRadius={"border-radius-s"}
      $ba="border-solid-s"
      $pa={"inner-padding-m"}
      $gap={"space-between-xs"}
      $flexDirection={"column"}
      $width={"100%"}
      $highlighted={!!highlighted}
      $disabled={!!disabled}
    >
      <OakFlex
        $justifyContent={"flex-start"}
        $alignItems={"center"}
        $width={"100%"}
        $gap={"space-between-xs"}
        $font={"heading-7"}
      >
        <OakCheckBox
          checked={checked}
          value={`Caption ID: ${captionId}`}
          displayValue=""
          disabled={disabled}
          onChange={onCheckChanged}
          aria-labelledby={captionId}
          aria-label={`check caption ${captionId}`}
          data-testid="checkbox"
          id={captionId}
        />
        <OakFlex
          $font={"heading-7"}
          $justifyContent={"flex-start"}
          $gap={"space-between-ssx"}
          $flexWrap={"wrap"}
        >
          <OakSpan $font={"heading-7"}>Caption ID: {captionId}</OakSpan>
          <OakSpan> â€¢ </OakSpan>
          <OakBox>Video Title: {videoTitle}</OakBox>
        </OakFlex>

        <OakFlex $flexGrow={10} $justifyContent={"flex-end"}>
          <InternalButton
            onClick={onEditClick}
            aria-label={`edit caption ${captionId} in rev`}
          >
            <OakFlex $alignItems={"center"} $gap={"space-between-sssx"}>
              Edit
              <StyledOakIcon
                iconWidth="all-spacing-6"
                iconHeight="all-spacing-6"
                alt=""
                iconName="external"
              />
            </OakFlex>
          </InternalButton>
        </OakFlex>
      </OakFlex>
      <OakFlex
        $justifyContent={"flex-start"}
        $alignItems={"center"}
        $width={"100%"}
        $gap={"all-spacing-7"}
        $font={"body-2"}
      >
        <InternalLink
          onClick={onLessonUidClick}
          color={"black"}
          visitedColor={"border-brand"}
          hoverColor={"black"}
          activeColor={"black"}
          disabledColor={"text-disabled"}
          aria-label={`view lesson ${lessonUid}`}
          data-testid="lesson_uid"
        >
          <OakFlex $gap={"space-between-sssx"}>
            {lessonUid}
            <StyledOakIcon
              iconWidth="all-spacing-6"
              iconHeight="all-spacing-6"
              alt=""
              iconName="external"
            />
          </OakFlex>
        </InternalLink>
        <OakFlex $alignItems={"center"} $gap={"space-between-sssx"}>
          {" "}
          {/* the video icon has no natural padding so whilst inconsistent this looks better */}
          <StyledOakIcon alt="" iconName="video" />
          {getVideoTypeText(videoType)}
        </OakFlex>
        {lastEdited ? (
          <OakFlex $alignItems={"center"}>
            <StyledOakIcon alt="" iconName="equipment-required" />
            Edited {getTimeText(lastEdited)}
          </OakFlex>
        ) : null}
        <OakFlex $alignItems={"center"}>
          <StyledOakIcon alt="" iconName="success" />
          Updated {getTimeText(lastUpdated)}
        </OakFlex>
      </OakFlex>
    </StyledFlexBox>
  );
};

function getVideoTypeText(videoType: string): string {
  switch (videoType) {
    case "lesson":
      return "Lesson video";
    case "media clip":
      return "Media clip";
    default:
      return "Unknown";
  }
}
