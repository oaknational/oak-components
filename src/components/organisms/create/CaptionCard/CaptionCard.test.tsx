import "@testing-library/jest-dom";
import React from "react";
import { create } from "react-test-renderer";
import { act } from "@testing-library/react";

import { CaptionCard, getTimeText } from "./CaptionCard";

import renderWithTheme from "@/test-helpers/renderWithTheme";
import { OakThemeProvider } from "@/components/atoms";
import { oakDefaultTheme } from "@/styles";

describe("CaptionCard", () => {
  it("renders a caption card", () => {
    const { getByTestId } = renderWithTheme(
      <CaptionCard
        captionId={"CAP-TEST-01234"}
        videoTitle={"This is a test video title"}
        lessonUid={"LESS-TEST-01234"}
        videoType={"lesson"}
        lastUpdated={"2023-01-01"}
        lastEdited={"2023-01-01"}
        checked={false}
        highlighted={false}
        onCheckChanged={() => {}}
        disabled={false}
      />,
    );
    expect(getByTestId("caption-card")).toBeInTheDocument();
  });

  it("matches snapshot", () => {
    const tree = create(
      <OakThemeProvider theme={oakDefaultTheme}>
        <CaptionCard
          captionId={"CAP-TEST-01234"}
          videoTitle={"This is a test video title"}
          lessonUid={"LESS-TEST-01234"}
          videoType={"lesson"}
          lastUpdated={"2023-01-01"}
          lastEdited={"2023-01-01"}
          checked={false}
          highlighted={false}
          onCheckChanged={() => {}}
          disabled={false}
        />
        ,
      </OakThemeProvider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("does not include the last edited time if one is not provided", () => {
    const { queryByText } = renderWithTheme(
      <CaptionCard
        captionId={"CAP-TEST-01234"}
        videoTitle={"This is a test video title"}
        lessonUid={"LESS-TEST-01234"}
        videoType={"lesson"}
        lastUpdated={"2023-01-01"}
        checked={true}
        highlighted={false}
        onCheckChanged={() => {}}
        disabled={false}
      />,
    );
    expect(queryByText("Edited")).not.toBeInTheDocument();
  });

  it("calls onChangeChecked when checkbox is clicked", () => {
    const mockCallback = jest.fn();
    const { queryByTestId } = renderWithTheme(
      <CaptionCard
        captionId={"CAP-TEST-01234"}
        videoTitle={"This is a test video title"}
        lessonUid={"LESS-TEST-01234"}
        videoType={"lesson"}
        lastUpdated={"2023-01-01"}
        checked={false}
        highlighted={false}
        onCheckChanged={mockCallback}
        disabled={false}
      />,
    );
    queryByTestId("checkbox")?.click();
    expect(mockCallback).toHaveBeenCalled();
  });
  it("calls onLessonUidClicked when lesson uid is clicked", () => {
    const mockCallback = jest.fn();
    const { queryByTestId } = renderWithTheme(
      <CaptionCard
        captionId={"CAP-TEST-01234"}
        videoTitle={"This is a test video title"}
        lessonUid={"LESS-TEST-01234"}
        videoType={"lesson"}
        lastUpdated={"2023-01-01"}
        checked={false}
        highlighted={false}
        onLessonUidClick={mockCallback}
        disabled={false}
      />,
    );
    queryByTestId("lesson_uid")?.click();
    expect(mockCallback).toHaveBeenCalled();
  });

  it("should display checked box if checked prop is true", () => {
    const { queryByTestId } = renderWithTheme(
      <CaptionCard
        captionId={"CAP-TEST-01234"}
        videoTitle={"This is a test video title"}
        lessonUid={"LESS-TEST-01234"}
        videoType={"lesson"}
        lastUpdated={"2023-01-01"}
        checked={true}
        highlighted={false}
        onCheckChanged={() => {}}
        disabled={false}
      />,
    );
    expect(queryByTestId("checkbox")).toBeInTheDocument();
    expect(queryByTestId("checkbox")).toBeChecked();
  });

  it("should display unchecked box if checked prop is false", () => {
    const { queryByTestId } = renderWithTheme(
      <CaptionCard
        captionId={"CAP-TEST-01234"}
        videoTitle={"This is a test video title"}
        lessonUid={"LESS-TEST-01234"}
        videoType={"lesson"}
        lastUpdated={"2023-01-01"}
        checked={false}
        highlighted={false}
        onCheckChanged={() => {}}
        disabled={false}
      />,
    );
    expect(queryByTestId("checkbox")).toBeInTheDocument();
    expect(queryByTestId("checkbox")).not.toBeChecked();
  });

  it("should use a different background if highlighted prop is true", () => {
    renderWithTheme(
      <CaptionCard
        captionId={"CAP-TEST-01234"}
        videoTitle={"This is a test video title"}
        lessonUid={"LESS-TEST-01234"}
        videoType={"lesson"}
        lastUpdated={"2023-01-01"}
        checked={false}
        highlighted={false}
        onCheckChanged={() => {}}
        disabled={false}
        data-testid="caption-card-original"
      />,
    );
    const { queryByTestId } = renderWithTheme(
      <CaptionCard
        captionId={"CAP-TEST-01234"}
        videoTitle={"This is a test video title"}
        lessonUid={"LESS-TEST-01234"}
        videoType={"lesson"}
        lastUpdated={"2023-01-01"}
        checked={false}
        highlighted={true}
        onCheckChanged={() => {}}
        disabled={false}
        data-testid="caption-card"
      />,
    );
    // expect(queryByTestId("caption-card")).toBeInTheDocument();
    expect(queryByTestId("caption-card")?.attributes).not.toBe(
      queryByTestId("caption-card-original")?.attributes,
    );
  });

  it("should render the caption information", () => {
    const { queryByText } = renderWithTheme(
      <CaptionCard
        captionId={"test-caption-id"}
        videoTitle={"test-video-title"}
        lessonUid={"test-lesson-uid"}
        videoType={"lesson"}
        lastUpdated={new Date().toUTCString()}
        checked={false}
        highlighted={false}
        onCheckChanged={() => {}}
        disabled={false}
      />,
    );
    expect(
      queryByText("test-caption-id", { exact: false }),
    ).toBeInTheDocument();
    expect(
      queryByText("test-video-title", { exact: false }),
    ).toBeInTheDocument();
    expect(
      queryByText("test-lesson-uid", { exact: false }),
    ).toBeInTheDocument();
    expect(queryByText("Updated", { exact: false })).toBeInTheDocument();
  });

  it("clicking checkbox triggers onChangeChecked", async () => {
    const onCheckChanged = jest.fn();
    const { getByTestId } = renderWithTheme(
      <CaptionCard
        captionId={"CAP-TEST-01234"}
        videoTitle={"This is a test video title"}
        lessonUid={"LESS-TEST-01234"}
        videoType={"lesson"}
        lastUpdated={"2023-01-01"}
        onCheckChanged={onCheckChanged}
        checked={false}
        disabled={false}
      />,
    );
    act(() => {
      getByTestId("checkbox").click();
    });
    expect(getByTestId("checkbox")).toBeInTheDocument();
    expect(onCheckChanged).toHaveBeenCalled();
  });

  it("checkbox is not clickable when the card is disabled", () => {
    const onCheckChanged = jest.fn();
    const { getByTestId } = renderWithTheme(
      <CaptionCard
        captionId={"CAP-TEST-01234"}
        videoTitle={"This is a test video title"}
        lessonUid={"LESS-TEST-01234"}
        videoType={"lesson"}
        lastUpdated={"2023-01-01"}
        onCheckChanged={onCheckChanged}
        checked={false}
        disabled={true}
      />,
    );
    act(() => {
      getByTestId("checkbox").click();
    });
    expect(onCheckChanged).not.toHaveBeenCalled();
  });
});

// May need to increase this depending on the test run time
const timeToSubtract = 10;

describe("getTimeText", () => {
  it("returns 'just now' if time is less than a second in the past", () => {
    expect(getTimeText(new Date().toUTCString())).toBe("just now");
  });
  it("returns time in secs if time is less than a minute in the past", () => {
    expect(getTimeText(Date.now() - (1000 * 60 - timeToSubtract))).toBe(
      "59 secs ago",
    );
  });
  it("returns time in mins if time is less than an hour in the past", () => {
    expect(getTimeText(Date.now() - (1000 * 60 * 60 - timeToSubtract))).toBe(
      "59 mins ago",
    );
  });
  it("returns time in hours if time is less than a day in the past", () => {
    expect(
      getTimeText(Date.now() - (1000 * 60 * 60 * 24 - timeToSubtract)),
    ).toBe("23 hrs ago");
  });
  it("returns time in days if time is less than a week in the past", () => {
    expect(
      getTimeText(Date.now() - (1000 * 60 * 60 * 24 * 7 - timeToSubtract)),
    ).toBe("6 days ago");
  });
  it("returns time in weeks if time is less than a month in the past", () => {
    expect(
      getTimeText(Date.now() - (1000 * 60 * 60 * 24 * 30 - timeToSubtract)),
    ).toBe("4 weeks ago");
  });
  it("returns time in months if time is less than a year in the past", () => {
    expect(
      getTimeText(Date.now() - (1000 * 60 * 60 * 24 * 364 - timeToSubtract)),
    ).toBe("12 months ago");
  });
  it("returns time in years if time is more than a year in the past", () => {
    expect(getTimeText(Date.now() - 1000 * 60 * 60 * 24 * 365 * 2)).toBe(
      "2 years ago",
    );
  });
  it("returns 'at an unknown time' if the date is invalid", () => {
    expect(getTimeText("invalid date")).toBe("at an unknown time");
  });
});
