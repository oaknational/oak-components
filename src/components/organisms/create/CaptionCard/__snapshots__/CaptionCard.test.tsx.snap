// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CaptionCard matches snapshot 1`] = `
[
  .c0 {
  position: relative;
  width: 100%;
  min-height: 2.5rem;
  padding-left: 0.75rem;
  padding-right: 0.75rem;
  padding-top: 0.75rem;
  padding-bottom: 0.75rem;
  border: 0.063rem solid;
  border-radius: 0.25rem;
  font-family: --var(google-font),Lexend,sans-serif;
}

.c3 {
  width: 100%;
  font-family: --var(google-font),Lexend,sans-serif;
}

.c7 {
  position: relative;
  width: 1.5rem;
  height: 1.5rem;
  font-family: --var(google-font),Lexend,sans-serif;
}

.c13 {
  position: absolute;
  top: 0rem;
  left: 0rem;
  width: 1.5rem;
  height: 1.5rem;
  padding: 0rem;
  font-family: --var(google-font),Lexend,sans-serif;
}

.c15 {
  position: relative;
  width: 100%;
  min-width: 100%;
  height: 100%;
  min-height: 100%;
  font-family: --var(google-font),Lexend,sans-serif;
}

.c17 {
  font-family: --var(google-font),Lexend,sans-serif;
}

.c21 {
  position: relative;
  width: 1.5rem;
  min-width: 1.5rem;
  height: 1.5rem;
  min-height: 1.5rem;
  font-family: --var(google-font),Lexend,sans-serif;
}

.c25 {
  position: relative;
  min-height: 2.5rem;
  font-family: --var(google-font),Lexend,sans-serif;
}

.c27 {
  position: relative;
  width: 2rem;
  min-width: 2rem;
  height: 2rem;
  min-height: 2rem;
  font-family: --var(google-font),Lexend,sans-serif;
}

.c16 {
  -webkit-filter: invert(98%) sepia(98%) saturate(0%) hue-rotate(328deg) brightness(102%) contrast(102%);
  filter: invert(98%) sepia(98%) saturate(0%) hue-rotate(328deg) brightness(102%) contrast(102%);
  object-fit: contain;
}

.c22 {
  object-fit: contain;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 0.25rem;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  gap: 0.75rem;
}

.c18 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  -webkit-box-flex: 10;
  -webkit-flex-grow: 10;
  -ms-flex-positive: 10;
  flex-grow: 10;
}

.c20 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 0.25rem;
}

.c26 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c24 {
  font-family: --var(google-font),Lexend,sans-serif;
}

.c5 {
  font-family: --var(google-font),Lexend,sans-serif;
  color: #222222;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 1rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c19 {
  background: none;
  color: inherit;
  border: none;
  padding: 0;
  font: inherit;
  cursor: pointer;
  text-align: left;
  font-family: unset;
  outline: none;
  font-family: --var(google-font),Lexend,sans-serif;
}

.c19:disabled {
  pointer-events: none;
  cursor: default;
}

.c9 {
  -webkit-appearance: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  cursor: pointer;
  margin: 0;
  outline: none;
  border: 0.125rem solid;
  border-color: #222222;
  border-radius: 0.125rem;
  width: 1.5rem;
  height: 1.5rem;
  border-color: #808080;
}

.c9:checked {
  background: #222222;
  border-color: #222222;
}

.c9:disabled {
  pointer-events: none;
}

.c11:focus-visible {
  box-shadow: 0 0 0 0.125rem rgba(255,229,85,100%);
}

.c14 {
  pointer-events: none;
  opacity: 0;
}

input:checked + .c12 {
  opacity: 1;
}

.c23 {
  display: inline;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 0.25rem;
  outline: none;
  border-radius: 0.375rem;
  padding: 0.25rem;
  margin: -0.25rem;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  font: inherit;
  background: none;
  border: none;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  cursor: pointer;
  color: #222222;
}

.c23:focus-visible {
  box-shadow: 0 0 0 0.125rem rgba(255,229,85,100%), 0 0 0 0.3rem rgba(87,87,87,100%);
}

.c23:visited {
  color: #287c34;
}

.c23:active {
  color: #222222;
}

.c23[disabled] {
  cursor: not-allowed;
  color: #808080;
}

.c2:has(input:not(:disabled)) {
  cursor: default;
}

.c2:has(input:disabled) {
  pointer-events: none;
  cursor: none;
}

.c2:focus-within {
  box-shadow: 0 0 0 0.125rem rgba(255,229,85,100%), 0 0 0 0.3rem rgba(87,87,87,100%);
}

.c2:hover {
  background-color: #e4e4e4;
}

@media (hover:hover) {
  .c9:hover:not(:disabled) {
    border-color: #222222;
  }
}

@media (hover:hover) {
  .c10:hover:not(.c8:checked):not(.c8:disabled)::after {
    content: "";
    position: absolute;
    top: 50%;
    left: 50%;
    width: 60%;
    height: 60%;
    -webkit-transform: translate(-50%,-50%);
    -ms-transform: translate(-50%,-50%);
    transform: translate(-50%,-50%);
    border-radius: 0.125rem;
    background: #222222;
  }
}

@media (hover:hover) {
  .c23:hover,
  .c23:visited:hover {
    color: #222222;
  }
}

<div
    className="c0 c1 c2"
    data-testid="caption-card"
  >
    <div
      className="c3 c4"
    >
      <label
        className="c5 c6"
        disabled={false}
        htmlFor="CAP-TEST-01234"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <div
          className="c7"
        >
          <input
            aria-label="check caption CAP-TEST-01234"
            aria-labelledby="CAP-TEST-01234"
            checked={false}
            className="c8 c9 c10 c11"
            data-testid="checkbox"
            disabled={false}
            id="CAP-TEST-01234"
            name="CAP-TEST-01234"
            onChange={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            type="checkbox"
            value="Caption ID: CAP-TEST-01234"
          />
          <div
            className="c12 c13 c14"
          >
            <div
              className="c15"
            >
              <img
                alt=""
                className="c16"
                data-nimg="fill"
                decoding="async"
                loading="lazy"
                onError={[Function]}
                onLoad={[Function]}
                src="https://res.cloudinary.com/mock-cloudinary-cloud/image/upload/v1699954310/icons/efd3esaor6zqk7seh6kt.svg"
                style={
                  {
                    "bottom": 0,
                    "color": "transparent",
                    "height": "100%",
                    "left": 0,
                    "objectFit": undefined,
                    "objectPosition": undefined,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "width": "100%",
                  }
                }
              />
            </div>
          </div>
        </div>
        Caption ID: CAP-TEST-01234
      </label>
      <div
        className="c17"
      >
        Video Title: 
        This is a test video title
      </div>
      <div
        className="c17 c18"
      >
        <button
          aria-label="edit caption CAP-TEST-01234 in rev"
          className="c19"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="c17 c20"
          >
            Edit
            <div
              className="c21 "
            >
              <img
                alt=""
                className="c22"
                data-nimg="fill"
                decoding="async"
                loading="lazy"
                onError={[Function]}
                onLoad={[Function]}
                src="https://res.cloudinary.com/mock-cloudinary-cloud/image/upload/v1699953892/icons/hlxmejse3mcr4tqo6t8u.svg"
                style={
                  {
                    "bottom": 0,
                    "color": "transparent",
                    "height": "100%",
                    "left": 0,
                    "objectFit": undefined,
                    "objectPosition": undefined,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "width": "100%",
                  }
                }
              />
            </div>
          </div>
        </button>
      </div>
    </div>
    <div
      className="c3 c4"
    >
      <a
        aria-label="view lesson LESS-TEST-01234"
        className="c23"
        data-testid="lesson_uid"
      >
        <span
          className="c24"
        >
          <div
            className="c25 c20"
          >
            LESS-TEST-01234
            <div
              className="c21 "
            >
              <img
                alt=""
                className="c22"
                data-nimg="fill"
                decoding="async"
                loading="lazy"
                onError={[Function]}
                onLoad={[Function]}
                src="https://res.cloudinary.com/mock-cloudinary-cloud/image/upload/v1699953892/icons/hlxmejse3mcr4tqo6t8u.svg"
                style={
                  {
                    "bottom": 0,
                    "color": "transparent",
                    "height": "100%",
                    "left": 0,
                    "objectFit": undefined,
                    "objectPosition": undefined,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "width": "100%",
                  }
                }
              />
            </div>
          </div>
        </span>
      </a>
      <div
        className="c17 c26"
      >
        <div
          className="c27 "
        >
          <img
            alt=""
            className="c22"
            data-nimg="fill"
            decoding="async"
            loading="lazy"
            onError={[Function]}
            onLoad={[Function]}
            src="https://res.cloudinary.com/mock-cloudinary-cloud/image/upload/v1705416078/icons/wzey1zfxrvv3apeebbf5.svg"
            style={
              {
                "bottom": 0,
                "color": "transparent",
                "height": "100%",
                "left": 0,
                "objectFit": undefined,
                "objectPosition": undefined,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "width": "100%",
              }
            }
          />
        </div>
        Lesson video
      </div>
      <div
        className="c17 c26"
      >
        <div
          className="c27 "
        >
          <img
            alt=""
            className="c22"
            data-nimg="fill"
            decoding="async"
            loading="lazy"
            onError={[Function]}
            onLoad={[Function]}
            src="https://res.cloudinary.com/mock-cloudinary-cloud/image/upload/v1699953925/icons/pw22bdhj2vrzfv2ogi4e.svg"
            style={
              {
                "bottom": 0,
                "color": "transparent",
                "height": "100%",
                "left": 0,
                "objectFit": undefined,
                "objectPosition": undefined,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "width": "100%",
              }
            }
          />
        </div>
        Edited 
        2 years ago
      </div>
      <div
        className="c17 c26"
      >
        <div
          className="c27 "
        >
          <img
            alt=""
            className="c22"
            data-nimg="fill"
            decoding="async"
            loading="lazy"
            onError={[Function]}
            onLoad={[Function]}
            src="https://res.cloudinary.com/mock-cloudinary-cloud/image/upload/v1699895534/icons/Icon-Success_aiiprx.svg"
            style={
              {
                "bottom": 0,
                "color": "transparent",
                "height": "100%",
                "left": 0,
                "objectFit": undefined,
                "objectPosition": undefined,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "width": "100%",
              }
            }
          />
        </div>
        Updated 
        2 years ago
      </div>
    </div>
  </div>,
  ",",
]
`;
